### ISSUE TITLE
Cannot fit requested classes in a single dex file with 3 package

### ISSUE THREAD

**Issue by ikaro88** [firebase/quickstart-unity#664]
### Please fill in the following fields:
Unity editor version: 2019.1.9
Firebase Unity SDK version: Last
Source you installed the SDK : unitypackage
Firebase plugins in use (Auth, Database, and analytics.):
Additional SDKs you are using (Facebook, AdMob): 
Platform you are using the Unity editor on (windows): 
Platform you are targeting (adroid): 
Scripting Runtime (IL2CPP):

### Please describe the issue here:
When i compile and I put all the 3 package (database, analitics and auth) i get this error on compiling: "Cannot fit requested classes in a single dex file"

### Please answer the following, if applicable:
i just added the 3 package and compile, with only 2 of them i have no probelm

What's the issue repro rate? (eg 100%, 1/5 etc)
100%

**Comment by gromilQaaaa** [comment]
You need to use multidex in your mainTemplate gradle file.
https://wiki.appodeal.com/en/unity/enable-multidex-for-unity-project

**Comment by chkuang-g** [comment]
@gromilQaaaa Thank you for offering the solution.

@ikaro88 
It does looks like an issue Multidex is trying to solve. Quick question: are you currently using Custom Gradle Templete, a.k.a, mainTemplate.gradle?  Does it work if you don't use it?

**Comment by chkuang-g** [comment]
Hey folks,

Another way to solve this to turn on `Proguard`, which stripes away unused symbols.

To do so, bring up `Player Settings`.  Under `Publishing Settings` tab, looking for `Minify`.
- Switch `Release` to `Proguard` for release build
- Switch `Debug` to `Proguard` for debug build

In this case, you should not need multidex.

**Comment by nmill99** [comment]
@chkuang-g your Minify with Proguard suggestion fixed this problem for me.  I no longer get this error on build.

FYI - I just added Firebase > Firestore (no other Firebase packs) to my project for the first time and did the first build when I got this error.  I'm not familiar with multidex so I don't think I had anything custom there.

Unity editor version: 2019.4.0f1
Firebase Unity SDK version: 6.15.1
Source you installed the SDK : unitypackage
Firebase plugins in use (Auth, Database, and analytics.): Firebase only
Additional SDKs you are using: none
Platform you are using the Unity editor on: windows
Platform you are targeting: android
Scripting Runtime: not an option
